### ISSUE TITLE
failed to memset memory: HIP_ERROR_InvalidValue

### ISSUE THREAD

**Issue by juliusshufan** [ROCmSoftwarePlatform/tensorflow-upstream#759]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS: Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version: 1.15 rocm
- Python version: 3.6
- ROCm/MIOpen version:
- GPU model and memory: 5700XT 

**Describe the problem**
I am following the guild on https://rocm.github.io/install.html#rocm-binary-package-structure to install the ROCm related libs and tool chains. So far below command run properly,
/opt/rocm/bin/rocminfo
/opt/rocm/opencl/bin/x86_64/clinfo

However, on running a simple tensorflow example:
import tensorflow as tf
mnist = tf.keras.datasets.mnist

(x_train, y_train),(x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(),
  tf.keras.layers.Dense(512, activation=tf.nn.relu),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=5)
model.evaluate(x_test, y_test)

Error happens:
2019-12-18 16:10:46.391155: E tensorflow/stream_executor/rocm/rocm_driver.cc:506] failed to memset memory: HIP_ERROR_InvalidValue
Segmentation fault (core dumped)

**Any other info / logs**
Full log:
2019-12-18 16:10:40.624634: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libhip_hcc.so
2019-12-18 16:10:40.690714: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1629] Found device 0 with properties:
name: Device 731f
AMDGPU ISA: gfx1010
memoryClockRate (GHz) 0.1
pciBusID 0000:67:00.0
2019-12-18 16:10:40.722320: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocblas.so
2019-12-18 16:10:40.723078: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libMIOpen.so
2019-12-18 16:10:40.724007: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocfft.so
2019-12-18 16:10:40.724174: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocrand.so
2019-12-18 16:10:40.724237: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2019-12-18 16:10:40.724495: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
2019-12-18 16:10:40.736117: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2600000000 Hz
2019-12-18 16:10:40.739055: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55a3203cbbf0 executing computations on platform Host. Devices:
2019-12-18 16:10:40.739095: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Host, Default Version
2019-12-18 16:10:40.739289: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1629] Found device 0 with properties:
name: Device 731f
AMDGPU ISA: gfx1010
memoryClockRate (GHz) 0.1
pciBusID 0000:67:00.0
2019-12-18 16:10:40.739340: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocblas.so
2019-12-18 16:10:40.739362: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libMIOpen.so
2019-12-18 16:10:40.739384: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocfft.so
2019-12-18 16:10:40.739404: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library librocrand.so
2019-12-18 16:10:40.739468: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1746] Adding visible gpu devices: 0
2019-12-18 16:10:40.739582: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-12-18 16:10:40.739610: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0
2019-12-18 16:10:40.739632: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N
2019-12-18 16:10:40.739804: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 7524 MB memory) -> physical GPU (device: 0, name: Device 731f, pci bus id: 0000:67:00.0)
2019-12-18 16:10:46.391155: E tensorflow/stream_executor/rocm/rocm_driver.cc:506] failed to memset memory: HIP_ERROR_InvalidValue
Segmentation fault (core dumped)

**Comment by sunway513** [comment]
Hi @juliusshufan , ROCm currently doesn't have full GFX10 architecture support. We'll update this thread when we have more information to share.

**Comment by juliusshufan** [comment]
@sunway513 Thanks for the information, is GFX10 support on the ROCm roadmap?

**Comment by sunway513** [comment]
Hi @juliusshufan , I don't have more information to share at the moment. 
For ROCm support status, please check with the ROCm main repository below:
https://github.com/RadeonOpenCompute/ROCm
